proto/substrait/algebra.proto

```diff
@@ -825,9 +825,11 @@ message Expression {
       UserDefined user_defined = 33;
     }
 
-    // whether the literal type should be treated as a nullable type. Applies to
-    // all members of union other than the Typed null (which should directly
-    // declare nullability).
+    // Whether the literal_type above should be treated as a nullable type.
+    // Applies to all members of the literal_type oneof EXCEPT:
+    //  * Type null             (must be nullable by definition)
+    //  * Type.List empty_list  (use Type.List::nullability)
```

Comment:
If the Type.List::nullability is the nullability of the type inside the list then I don't think this applies.

Reply:
That matches what I believe.  But for empty lists and empty maps why is that any different than populated lists and maps?

Reply:
Empty lists and empty maps have two spots for list nullability (so we need to pick one to be authoritative):

```
// List nullability (authoritative)
Literal::empty_list::list::nullability
// Also list nullability (redundant)
Literal::nullability
```

Populated lists and maps have one spot for list nullability and one spot for item nullability:

```
// Applies to the item
Literal::list::values...nullability
// Applies to the list (only option, thus authoritative)
Literal::nullability
```

Reply:
Okay, I see it now.  An alternative would be to change the type of of empty_list and empty_map to Type but that doesn't work simply for maps (which would need two nullability values for its key and type).